### PR TITLE
pmp: add PMP struct

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

The PMP wasn't working on the arty because the configure function wasn't actually iterating the PMP regions. This separates out the PMP from the config, and makes a lot more sense to me. And it works on the arty-e21 board.

### Testing Strategy

This pull request was tested by running on arty e21.

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: arch/rv32i/src/pmp.rs [hunk 1]
```diff
@@ -136,37 +152,29 @@ impl PMPRegion {
     }
 }
 
-pub trait PMPConfigType: Default + Copy + Clone + Sized {}
-
 /// Struct storing region configuration for RISCV PMP.
-pub struct PMPConfig<N: PMPConfigType> {
+pub struct PMPConfig {
```
Comment (alistair23) [inline arch/rv32i/src/pmp.rs:156]: This change is fine, but just noting that it is moving us further towards the macro approach as we get further away from the array generators.

File: arch/rv32i/src/pmp.rs [hunk 2]
```diff
@@ -136,37 +152,29 @@ impl PMPRegion {
...
-impl PMPConfigType for [Option<PMPRegion>; $x / 2] {}
-
-impl Default for PMPConfig<[Option<PMPRegion>; $x / 2]> {
+impl Default for PMPConfig {
     fn default() -> Self {
         PMPConfig {
             regions: [None; $x / 2],
             is_dirty: Cell::new(true),
-            last_configured_for: MapCell::empty(),
+
```
Comment (alistair23) [inline arch/rv32i/src/pmp.rs:171]: There is an extra new line here

File: arch/rv32i/src/pmp.rs (hunk 1 above)
Comment (hudson-ayers) [reply arch/rv32i/src/pmp.rs:156]: For general awareness, the minimum subset of const generics that will be stabilized now has its own feature gate (https://github.com/rust-lang/rust/pull/74877). Its not stable yet, but clearly on the fast track now.

File: arch/rv32i/src/pmp.rs (hunk 2 above)
Comment (bradjc) [inline arch/rv32i/src/pmp.rs:171]:
```suggestion
```

**Pull request merged**